File: 11286 Conformity/solution.java
import java.io.*;
import java.util.*;

public class Main {
    static short[] courses = new short[5];;
    public static void main(String[] args) throws IOException {
	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	    StreamTokenizer st = new StreamTokenizer(br);
	    int n;
	    StringBuilder output = new StringBuilder();
	    while(st.nextToken() == StreamTokenizer.TT_NUMBER && (n = (int)st.nval) != 0) {
	        long[] combinations = new long[n];
    		for(int i = 0; i < n; ++i) {
    		    for(int j = 0; j < 5; ++j) {
    		        st.nextToken();
    		        courses[j] = (short)st.nval;
    		    }
    		    Arrays.sort(courses);
    		    for(int j = 0; j < 5; ++j)
    		        combinations[i] = 1000 * combinations[i] + courses[j];
    		}
    		Arrays.sort(combinations);
    		int max = 1, students = 0, count = 1;
    		for(int i = 1; i < n; ++i) {
    		    if(combinations[i] == combinations[i - 1])
    		        ++count;
    		    else {
    		        if(max < count)
        		        students = max = count;
        		    else if(max == count)
        		        students += max;
        		    count = 1;
    		    }
    		}
    		if(max < count)
            	students = max = count;
            else if(max == count)
                students += max;
    	    output.append(students).append('\n');
	    }
	    System.out.print(output);
	}
}

File: 11286 Conformity/solution.cpp
#include <bits/stdc++.h>
using namespace std;

array<short, 5> courses;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    int n;
    while(cin >> n && n != 0) {
        unordered_map<long, int> combinations;
        while(n--) {
            for(int i = 0; i < 5; ++i)
                cin >> courses[i];
            sort(courses.begin(), courses.end());
            long temp = 0;
            for(short c : courses)
                temp = 1000 * temp + c;
            ++combinations[temp];
        }
        int max = 0, students = 0;
        for(auto it : combinations) {
            if(max < it.second)
                students = max = it.second;
            else if(max == it.second)
                students += max;
        }
        cout << students << '\n';
    }
    return 0;
}
